apps/examples/testcase/le_tc/network/itc_net_setsockopt.c

```diff
@@ -0,0 +1,237 @@
+*
+* Copyright 2017 Samsung Electronics All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*
+****************************************************************************/
+
+/// @file itc_net_setsockopt.c
+/// @brief ITC Test Case Example for setsockopt() API
+
+#include <sys/types.h>
+#include <sys/socket.h>
+
+#include "tc_internal.h"
+
+/**
+* @fn                   :itc_net_setsockopt_n_invalid_opt_name
+* @brief                :setsockopt accept opt name
+* @Scenario             :sending invalid opt name
+* @API's covered        :setsockopt()
+* Preconditions         :socket file descriptor
+* Postconditions        :None
+*/
+static void itc_net_setsockopt_n_invalid_opt_name(int s)
+{
+	int ret = -1;
+	int optname = 16; //invalid opt name
+	void *optval = NULL;
+	socklen_t optlen = 0;
+
+	ret = setsockopt(s, SOL_SOCKET, optname, optval, optlen);
+	TC_ASSERT_EQ("setsockopt", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :itc_net_setsockopt_p_rcvbuf
+* @brief                :setsockopt setting flag SO_RCVBUF
+* @Scenario             :setsockopt setting flag SO_RCVBUF
+* @API's covered        :setsockopt()
+* Preconditions         :socket file descriptor
+* Postconditions        :None
+*/
+static void itc_net_setsockopt_p_rcvbuf(int s)
+{
+	int ret = -1;
+	int size = 1000;
+
+	ret = setsockopt(s, SOL_SOCKET, SO_RCVBUF, &size, size);
+	TC_ASSERT_EQ("setsockopt", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             :itc_net_setsockopt_p_reuseport
+* @brief                :setsockopt setting flag SO_REUSEPORT
+* @scenario             :setsockopt setting flag SO_REUSEPORT
+* @apicovered           :setsockopt()
+* @precondition         :socket file descriptor
+* @postcondition        :None
+*/
+static void itc_net_setsockopt_p_reuseport(int s)
+{
+	int ret = -1;
+	int option = 1;
+
+	ret = setsockopt(s, SOL_SOCKET, SO_REUSEPORT, (char *)&option, sizeof option);
+	TC_ASSERT_EQ("setsockopt", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             :itc_net_setsockopt_p_reuseaddr
+* @brief                :setsockopt setting flag SO_REUSEADDR
+* @scenario             :setsockopt setting flag SO_REUSEADDR
+* @apicovered           :setsockopt()
+* @precondition         :socket file descriptor
+* @postcondition        :None
+*/
+static void itc_net_setsockopt_p_reuseaddr(int s)
+{
+	int ret = -1;
+	int option = 1;
+
+	ret = setsockopt(s, SOL_SOCKET, SO_REUSEADDR, (char *)&option, sizeof option);
```

Comment:
Let's use bracket like others.
```
sizeof(option)
```

Same for all of sizeof.

Reply:
changes are done as per your suggestion. please review